src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/profiledetails/ProfileDetailsService.kt

```diff
@@ -65,7 +65,10 @@ class ProfileDetailsService(
     validateProfileCode(request.profileType, request.profileCode)
 
     val booking = findLatestBookingOrThrow(offenderNo)
-    val (profileDetails, created) = booking.getOrCreateProfile().getOrCreateProfileDetails(request)
+    val (profileDetails, created) =
+      booking
+        .also { it.getOrCreateProfile() }
+        .getOrCreateProfileDetails(request)
```

Comment:
Slightly concerned about a race condition here - will test before doing anything complicated to cater for it.